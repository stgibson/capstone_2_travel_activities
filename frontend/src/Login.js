import React from "react";
import BaseForm from "./BaseForm";

/**
 * Component for displaying form to log in
 * @param {Object} param0 
 * @returns JSX code for rendering login page
 */
const Login = ({ login }) => {
  // remove defaults for production
  const inputs = [
    {
      name: "username",
      label: "Username:",
      type: "text",
      default: "testuser"
    },
    { name: "password", label: "Password", type: "password", default: "cab" }
  ]

  return (
    <BaseForm
      title="Log In"
      inputs={ inputs }
      btnText="Log In"
      submitCallback={ login }
    />
  );
};

export default Login;