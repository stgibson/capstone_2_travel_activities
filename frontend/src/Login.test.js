import React from "react";
import { render } from "@testing-library/react";
import Login from "./Login";

const login = jest.fn();

it("renders without crashing", () => {
  render(<Login login={ login } />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<Login login={ login } />);

  expect(asFragment()).toMatchSnapshot();
});
