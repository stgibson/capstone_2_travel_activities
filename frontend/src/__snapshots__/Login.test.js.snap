// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <h2>
    Log In
  </h2>
  <form
    class=""
  >
    <div
      class="form-group"
    >
      <label
        class="form-label"
        for="username"
      >
        Username:
      </label>
      <input
        class="form-control"
        id="username"
        type="text"
        value="testuser"
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class="form-label"
        for="password"
      >
        Password
      </label>
      <input
        class="form-control"
        id="password"
        type="password"
        value="testing"
      />
    </div>
  </form>
  <button
    class="btn btn-primary"
    type="button"
  >
    Log In
  </button>
</DocumentFragment>
`;
